pre_build/cccp_index_pre_build_check.py

```diff
@@ -0,0 +1,162 @@
+    new_project[0]['project']['git_url'] = giturl
+    new_project[0]['project']['git_branch'] = gitbranch
+    rel_path = "/"
+    if gitpath and gitpath != "/":
+        rel_path = gitpath if gitpath.startswith("/") else (rel_path + gitpath)
+    new_project[0]['project']['rel_path'] = rel_path
+    new_project[0]['project']['jobs'] = ['cccp-rundotsh-job']
+
+    if 'rundotshargs' not in new_project[0]:
+        new_project[0]['project']['rundotshargs'] = ''
+    elif new_project[0]['project']['rundotshargs'] is None:
+        new_project[0]['project']['rundotshargs'] = ''
+
+    new_project[0]['project']['desired_tag'] = desiredtag
+    new_project[0]['project']['prebuild_script'] = prebuild_script
+    return new_project
+
+
+def get_projects_from_index(indexdlocation):
+    projects = []
+    for yamlfile in glob(indexdlocation + "/*.yml"):
+        if "index_template" not in yamlfile:
+            stream = open(yamlfile, 'r')
+            index_yaml = yaml.load(stream, Loader=yaml.BaseLoader)
+            stream.close()
+            # print index_yaml['Projects']
+
+            for project in index_yaml['Projects']:
+                if(project['git-url'] is None):
+                    continue
+                else:
+                    try:
+                        appid = project['app-id']
+                        jobid = project['job-id']
+                        giturl = project['git-url']
+                        gitpath = project['git-path'] \
+                            if (project['git-path'] is not None) else ''
+                        gitbranch = project['git-branch']
+                        try:
+                            desiredtag = project['desired-tag'] \
+                                if (project['desired-tag'] is not None) \
+                                else 'latest'
+                        except Exception:
+                            desiredtag = 'latest'
+
+                        desiredtag = str(desiredtag)
+
+                        new_proj = [{'project': {}}]
+
+                        appid = appid.replace(
+                            '_', '-').replace('/', '-').replace('.', '-')
+                        jobid = jobid.replace(
+                            '_', '-').replace('/', '-').replace('.', '-')
+
+                        # Set the pre-build data if available
+                        prebuild_script = 'none' if not project.get(
+                            'prebuild-script') else project['prebuild-script']
+
+                        # Check for the project only if it requires pre-build
+                        if(prebuild_script != 'none'):
+                            projects.append(
+                                projectify(
+                                    new_proj, appid, jobid, giturl, gitpath,
+                                    gitbranch, desiredtag,
+                                    prebuild_script)
+                            )
+                    except Exception as e:
```

Comment:
Few lines above in the same file, we're doing `except Exception`. If it makes sense, can we have uniformity in the way we're catching exceptions?

Reply:
I did not get, how do you mean by this.

Reply:
I remember seeing except Exception at one place and except Exception as e at other. I was pointing to that inconsistency. But I don't see that anymore in this file.

Reply:
ok